internal/monitors/conviva/conviva.go

```diff
@@ -169,33 +168,35 @@ func (m *Monitor) fetchMetrics(contextTimeout time.Duration, semaphore chan stru
 				logger.Errorf("GET metric %s failed. %+v", metricConf.MetricParameter, err)
 				return
 			}
+			timestamp := time.Now()
 			dps := make([]*datapoint.Datapoint, 0)
 			for metricParameter, series := range res {
 				metricConf.logFilterStatuses(series.Meta.FiltersWarmup, series.Meta.FiltersNotExist, series.Meta.FiltersIncompleteData)
 				metricName := "conviva." + metricParameter
 				for filterID, metricValues := range series.FilterIDValuesMap {
+				L:
 					for i, metricValue := range metricValues {
 						select {
 						case <-ctx.Done():
 							logger.Error(ctx.Err())
 							return
 						default:
-							dp := sfxclient.GaugeF(
-								metricName,
-								map[string]string{"account": metricConf.Account, "filter": metricConf.filtersMap[filterID]},
-								metricValue)
-							// Checking the type of series and setting dimensions accordingly
+							dp := sfxclient.GaugeF(metricName, map[string]string{"account": metricConf.Account, "filter": metricConf.filtersMap[filterID]}, metricValue)
```

Comment:
Make all of this that creates the datapoint into a separate function.  This function is doing too much.